#VRML_SIM R2023a utf8

EXTERNPROTO "./E-puck.proto"

PROTO E-puck-project [
    field SFVec3f            translation                  0 0 0                    # Is `Transform.translation`.
    field SFRotation         rotation                     0 0 1 0                  # Is `Transform.rotation`.
]
{
  DEF epuck E-puck {
    hidden position_0_0 174.60085038966824
    hidden position_0_1 164.47030519840314
    hidden rotation_1 0 -1 0 1.3283382113601827
    hidden rotation_2 0 1 0 1.1074872117338908
    translation IS translation
    rotation IS rotation
    controller "mapping"
    supervisor TRUE
    camera_fieldOfView 0.4
    camera_width 50
    camera_height 50
    turretSlot [
        InertialUnit {
        rotation 0 1 0 3.14
        }
        DEF front_ds DistanceSensor {
        translation 0 0 -0.045
        rotation 0 1 0 1.5708
        name "front_ds"
        recognitionColors [
            0 0 0
        ]
        lookupTable [
            0 0 0
            1.27 1.27 0
        ]
        }
        DEF right_ds DistanceSensor {
        translation 0.035 -0.01 -0.01
        name "right_ds"
        lookupTable [
            0 0 0
            1.27 1.27 0
        ]
        }
        DEF left_ds DistanceSensor {
        translation -0.035 -0.01 -0.01
        rotation 0 1 0 3.14159
        name "left_ds"
        lookupTable [
            0 0 0
            1.27 1.27 0
        ]
        }
        Camera {
        translation 0 0 0.12
        rotation 0 1 0 0
        name "camera1"
        recognitionColors [
            1 1 0
        ]
        fieldOfView 0.8726646259971648
        width 256
        height 256
        projection "spherical"
        zoom Zoom {
            maxFieldOfView 6.28
            minFieldOfView 0
        }
        recognition Recognition {
        }
        }
    ]
    }
}